lib/block_view.go

```diff
@@ -4082,10 +4086,27 @@ func (bav *UtxoView) ConnectBlock(
 				return nil, errors.Wrapf(err, "ConnectBlock: txn #%d should not connect based on "+
 					"TxnConnectStatusByIndex but err is nil", txIndex)
 			}
```

Comment:
I think you want to remove this `if err == nil` check from here. We haven't tried to connect hte txn yet, and so shouldn't check for any errors on it